pkg/plugins/k8s/k8s_plugin.go

```diff
@@ -127,24 +127,6 @@ func (p *K8sPlugin) OnNodeStateChange(old, new *sriovnetworkv1.SriovNetworkNodeS
 		}
 	}
 
-	// Check switchdev config
```

Comment:
K8s plugin is only checking the services now, can you reformat the condition on line 121 to be 
if !utils.IsSwitchdevModeSpec(new.Spec) {
    continue
}

Can you also remove the word "not" in the comment in line 120

Reply:
@Mmduh-483 I thought we still need the `p.servicesStateUpdate()` to set the right flags or content for K8s_plugin struct, isn't it? Otherwise, in line `p.updateTarget.needUpdate()` , it will always be false.

Reply:
In the current implementation:
- p.servicesStateUpdate runs when if condition `utils.IsSwitchdevModeSpec(new.Spec) ` is met, I suggest a simple refactoring for the if condtion

- At the start of `OnNOdeChange` p.updateTarget is reset to be all false `p.updateTarget.needUpdate()` is always false if there is no switchdev spec

Reply:
So your suggestion is to change:
```
        if utils.IsSwitchdevModeSpec(new.Spec) {
                // Check services
                err = p.servicesStateUpdate()
                if err != nil {
                        glog.Errorf("k8s-plugin OnNodeStateChange(): failed : %v", err)
                        return
                }
        }
```
To:
```
        if !utils.IsSwitchdevModeSpec(new.Spec) {
                continue
        }
        err = p.servicesStateUpdate()
        if err != nil {
                glog.Errorf("k8s-plugin OnNodeStateChange(): failed : %v", err)
                return
        }
```
Is it correct?

Reply:
yes

Reply:
Done.